### PR TITLE
Get CircleCI lint check check working, then pass tests

### LINKED ISSUE #3: Set-up CI
- [ ] linting #11 
  - [ ] `rustfmt`
  - [ ] `clippy`
  - [ ] anything else?
- [ ] automated test run
- anything else?
- steal the setup from [fe](https://github.com/ethereum/fe)?

### PR DESCRIPTION
**Pull request by carver**
Fixes #3 

Todo:
- [x] `rustfmt`
- [x] build check?
- [x] squash

### CODE REVIEW

**Comment by carver** [comment]
There weren't any formatting errors to test against, but I temporarily pushed a commit with one to confirm that CI fails.

**Comment by carver** [comment]
After this, I'll actually start running the project tests (which seem to be failing!).

**Comment by carver** [comment]
I didn't do anything manual with "Cargo.lock", just ran a `cargo update`.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,8 +1,44 @@
 version: 2.1
 orbs:
-  rust: circleci/rust@1.50.0-beta.8
+  rust: circleci/rust@1.1.1
+jobs:
+  lint-build-test:
+        description: |
+            Check linting with Clippy, then build the crate.
```
Comment (njgheorghita) [inline .circleci/config.yml:7]: _nitpick_ `... then build the crate and run tests.`

**Review by njgheorghita — APPROVED** [review]
Looks good! Are there clippy settings somewhere that I'm missing? I've run clippy locally and not able to reproduce the linting warnings. Not sure why, but as long as it's being caught in ci then it's probably something wrong with my local setup that I'll sus out eventually

**Comment by carver** [comment]
> Are there clippy settings somewhere that I'm missing?

Maybe the version of rust (v1.51.0)? The VM seems to install clippy normally: `rustup component add clippy`.

**Pull request merged**